components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageFacetImpl.java

```diff
@@ -185,21 +199,29 @@ public void registerWritePolicySelector(final WritePolicySelector writePolicySel
   @Override
   @Guarded(by = STARTED)
   public StorageTx openTx() {
-    return openStorageTx(null);
+    return openStorageTx(databaseInstanceProvider.get().acquire(), false);
   }
 
   @Override
   @Guarded(by = STARTED)
   public StorageTx openTx(final ODatabaseDocumentTx db) {
     checkNotNull(db);
-    return openStorageTx(db);
+    return openStorageTx(db, true);
   }
 
-  private StorageTx openStorageTx(final ODatabaseDocumentTx db) {
-    ODatabaseDocumentTx database = db;
-    if (database == null) {
-      database = databaseInstanceProvider.get().acquire();
+  /**
+   * Returns the "principal name" to be used with current instance of {@link StorageTx}.
+   */
+  @Nonnull
+  private String txPrincipalName() {
+    ClientInfo clientInfo = clientInfoProvider.getCurrentThreadClientInfo();
+    if (clientInfo == null || clientInfo.getUserid() == null) {
+      return "system";
```

Comment:
This looks like something that could go into a utility/helper method in nexus-security for dealing with clientInfo.